internal/resources/app_endpoint_access_control_function.go

```diff
@@ -0,0 +1,315 @@
+package resources
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+
+	"github.com/hashicorp/terraform-plugin-framework/types"
+
+	"github.com/couchbasecloud/terraform-provider-couchbase-capella/internal/api"
+	"github.com/couchbasecloud/terraform-provider-couchbase-capella/internal/errors"
+	providerschema "github.com/couchbasecloud/terraform-provider-couchbase-capella/internal/schema"
+
+	"github.com/hashicorp/terraform-plugin-framework/path"
+	"github.com/hashicorp/terraform-plugin-framework/resource"
+	"github.com/hashicorp/terraform-plugin-log/tflog"
+)
+
+// Ensure the implementation satisfies the expected interfaces.
+var (
+	_ resource.Resource                = &AccessControlFunction{}
+	_ resource.ResourceWithConfigure   = &AccessControlFunction{}
+	_ resource.ResourceWithImportState = &AccessControlFunction{}
+)
+
+const errorUpsertingAccessFunction = "There was an error upserting the access control function.  Error: "
+
+// AccessControlFunction is the Access Control Function resource implementation.
+type AccessControlFunction struct {
+	*providerschema.Data
+}
+
+func NewAccessControlFunction() resource.Resource {
+	return &AccessControlFunction{}
+}
+
+// Metadata returns the access function resource type name.
+func (a *AccessControlFunction) Metadata(
+	_ context.Context, req resource.MetadataRequest, resp *resource.MetadataResponse,
+) {
+	resp.TypeName = req.ProviderTypeName + "_app_endpoint_access_control_function"
+}
+
+// Schema defines the schema for the access function resource.
+func (a *AccessControlFunction) Schema(_ context.Context, _ resource.SchemaRequest, resp *resource.SchemaResponse) {
+	resp.Schema = AccessControlFunctionSchema()
+}
+
+// Create upserts an access control function.
+func (a *AccessControlFunction) Create(ctx context.Context, req resource.CreateRequest, resp *resource.CreateResponse) {
+	var plan providerschema.AccessControlFunction
+	diags := req.Plan.Get(ctx, &plan)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	var (
+		organizationId = plan.OrganizationId.ValueString()
+		projectId      = plan.ProjectId.ValueString()
+		clusterId      = plan.ClusterId.ValueString()
+		appServiceId   = plan.AppServiceId.ValueString()
+		keyspace       = fmt.Sprintf(
+			"%s.%s.%s",
+			plan.AppEndpoint.ValueString(),
+			plan.Scope.ValueString(),
+			plan.Collection.ValueString(),
+		)
```

Comment:
I intentionally put this into a single keyspace field because if a user wanted to work with the default scope and collection they might omit the scope and collection and expect it to work the same way as the v4 api.
This would not work in that case, discussing with PM